### PR TITLE
chore: standard error state and message

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #3835 

### Description:

Changes:
- improve default message for Forbidden and Authorization errors
- create a generic component that handles all types of Lightdash errors

Error on space page
Screenshot 2023-01-03 at 16 51 43

Error on chart page
Screenshot 2023-01-03 at 16 52 13

Error on dashboard page
Screenshot 2023-01-03 at 16 52 54

### CODE REVIEW

File: packages/frontend/src/components/common/Error/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+const DEFAULT_ERROR_PROPS: ComponentProps<typeof NonIdealState> = {
+    icon: 'error',
+    title: 'Unexpected error',
+};
+
+const Error: FC<{ error: LightdashError }> = ({ error }) => {
+    const props = useMemo<ComponentProps<typeof NonIdealState>>(() => {
+        try {
+            console.log('error', error);
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Error/index.tsx:18]: extra

**Review by IrakliJani — COMMENTED** [review]
nice cleanup.
we use nonideal state with error message over and over in other places as well. maybe update those too?

CleanShot 2023-01-03 at 21 02 39@2x

**Review by TuringLovesDeathMetal — CHANGES_REQUESTED** [review]
@ZeRego looks great! thanks for improving this ❤️ 

Two things:
1. Will we use teh same dashboard/chart access error messages if one day we let people limit access to specific charts/dashboards (not just the spaces they're in)? If yes, then maybe we change the error message to: `You don't have access to this chart/dashboard` (instead of talking about the spaces they're in). If no, then I quite like how specific/helpful it is! So I'd keep it :) 
2. Can we change the first error message, it reads a bit funny 🙏 So, `You don't have access to this resource or action.` (instead of `you have no`...)

**Comment by ZeRego** [comment]
> @ZeRego looks great! thanks for improving this ❤️
> 
> Two things:
> 
> 1. Will we use teh same dashboard/chart access error messages if one day we let people limit access to specific charts/dashboards (not just the spaces they're in)? If yes, then maybe we change the error message to: `You don't have access to this chart/dashboard` (instead of talking about the spaces they're in). If no, then I quite like how specific/helpful it is! So I'd keep it :)
> 2. Can we change the first error message, it reads a bit funny 🙏 So, `You don't have access to this resource or action.` (instead of `you have no`...)

1. Yes, the messages will be different. 
2. fixed :)

**Review by TuringLovesDeathMetal — APPROVED** [review]
Nice! Looks great :)

**Review by IrakliJani — APPROVED** [review]
🚢 🚢 🚢

**Pull request merged**